rfcs/000-third-party-software-installation.md

```diff
@@ -0,0 +1,109 @@
+
+# Motivation
+
+Currently, V only allows downloading the source code for V modules in the `VMODULES` directory through the `v install` command.
+This is good for making modules globally available. There is currently no streamlined way of _installing_ or _fetching_ or _obtaining_ third-party software written in V.
+
+# Guide-level explanation
+
+Take [_klonol_](https://github.com/hungrybluedev/klonol) for example.
+It is a command-line tool written in V that helps to maintain a local copy of all git repositories on GitHub and Gitea.
+If you want to install it, you have to clone the source yourself, run `v build.vsh` and then add the `bin` folder to `PATH`.
+A nice thing to have is a `v fetch <repo>` or `v obtain <repo>` command (or something else, please comment) that will clone the repo and build it for you.
+
+Only once, like when V is install, for example, the user needs to add `~/.v/bin` to `PATH`.
+This will make all third-party applications installed using `v fetch <repo>` available from anywhere.
+
+# Reference-level explanation
+
+It will piggyback off of `v install`.
+The act of installing the software is done through the following additional steps:
+
+1. Running `v build.vsh` or some other specified build script in the root of the repo.
+2. This will create an executable, along with all other
+   relevant files (like templates, etc) in the `bin` directory of the repo. Ideally, the resources will be embedded in the executable to allow for only one executable to be enough.
+3. The executable will be added to `~/.v/bin` for the user to call from anywhere.
+   TODO.
+   An additional **suggestion** is to have the `~/.vmodules` directory be replaced in favour of `~/.v/modules`.
+
+If there is another script (other than `build.vsh`) that generates the executable
+
+A sample `build.vsh` script is:
+
+```text
+println('Removing old artifacts...')
+rmdir_all('bin') or {}
+println('Done removing "bin" directory.')
+
+println('\nCreating new output directory "bin"...')
+mkdir('bin') ?
+println('Done creating "bin" directory.')
+
+println('\nChecking if everything is formatted correctly...')
+execute_or_panic('v fmt -verify .')
+println('Done checking formatting.')
+
+println('\nCompiling and building executable...')
+execute_or_panic('v -prod . -o bin/klonol')
+println('Done compiling and placing executable in "bin".')
+```
+
+Taken from [klonol](https://github.com/hungrybluedev/klonol/blob/6e36e5676542aae40e0124de13055643484bb015/build.vsh).
+
+Users will have to add `~/.v/bin` to their `PATH` once.
+
+We also make use of the `bin` directory to keep the other files in the root of the repository out of the search path.
+
+# Drawbacks
+
+- Security concerns. Perhaps users should take some extra steps to install software from third-party sources.
+- It will confuse users between the `v install` and `v fetch` or `v obtain` commands.
+- Users need to tinker with their `PATH`.
+- Third-party projects with the same name will cause namespace collision.
```

Comment:
We could define the name of the exe in the `v.mod`. And we could prevent duplication on the V package manager website by checking if it exists already?

Reply:
Or we could just overwrite the old executable by the new one the user installed. The user will be notified before overwriting.

Reply:
in the case of binary it is true, but with library, we must do it carefully, some apps may be dependent from some old version